### PR TITLE
NAT cost optimization

### PR DESCRIPTION
**Pull request by arman-promo**
PR for this [issue](https://github.com/lean-delivery/tf-module-aws-core/issues/2)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -47,3 +47,33 @@ resource "aws_default_security_group" "assign-name" {
 
   tags = "${merge(local.default_tags, var.tags, map("Name", "${var.project}-${var.environment}"))}"
 }
+
+resource "aws_eip" "nat" {
+  count     = "${ var.enable_nat_gateway ? 0 : length(var.availability_zones) }"
```
Comment (ToROxI) [inline main.tf:52]: Have you tested this module? I have strong feeling that you'll get `value of 'count' cannot be computed`

Comment (arman-promo) [reply main.tf:52]: > Have you tested this module? I have strong feeling that you'll get `value of 'count' cannot be computed`

I have no experience in terraform tests, but I did "plan" and "apply", and review through console

Comment (ToROxI) [reply main.tf:52]: That is not enough. 
Please try to invoke this code as module
```
module "test-core" {
  ...
  ...
}
```

Comment (arman-promo) [reply main.tf:52]: Invoked with this code
```
#test.tf
provider "aws" {
  profile   = "dev"
  region    = "us-west-2"
}
variable "amis" {
  type = "map"
}
variable "region" {
  default = "us-west-2"
}

variable "instance_type" {
  default = "t2.micro"
}

module "vpc" {
    source  = "../tf-module-aws-core"

    project             = "project"
    environment         = "dev"
    availability_zones  = ["us-west-2a", "us-west-2b"]
    vpc_cidr            = "10.10.0.0/16"
    private_subnets    = ["10.10.1.0/24", "10.10.2.0/24"]
    public_subnets     = ["10.10.11.0/24", "10.10.12.0/24"]

    enable_nat_gateway  = "false"
    region              = "${var.region}"
    amis                = "${var.amis}"
    instance_type       = "${var.instance_type}"
}

#terraform.tfvars
amis = { us-west-2 = "ami-79eca201" }
```
plan logs:
```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + module.vpc.aws_default_security_group.assign-name
      id:                                   <computed>
      arn:                                  <computed>
      egress.#:                             "1"
      egress.482069346.cidr_blocks.#:       "1"
      egress.482069346.cidr_blocks.0:       "0.0.0.0/0"
      egress.482069346.description:         ""
      egress.482069346.from_port:           "0"
      egress.482069346.ipv6_cidr_blocks.#:  "0"
      egress.482069346.prefix_list_ids.#:   "0"
      egress.482069346.protocol:            "-1"
      egress.482069346.security_groups.#:   "0"
      egress.482069346.self:                "false"
      egress.482069346.to_port:             "0"
      ingress.#:                            "1"
      ingress.753360330.cidr_blocks.#:      "0"
      ingress.753360330.description:        ""
      ingress.753360330.from_port:          "0"
      ingress.753360330.ipv6_cidr_blocks.#: "0"
      ingress.753360330.prefix_list_ids.#:  "0"
      ingress.753360330.protocol:           "-1"
      ingress.753360330.security_groups.#:  "0"
      ingress.753360330.self:               "true"
      ingress.753360330.to_port:            "0"
      name:                                 <computed>
      owner_id:                             <computed>
      revoke_rules_on_delete:               "false"
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev"
      tags.Project:                         "project"
      vpc_id:                               "${module.vpc.vpc_id}"

  + module.vpc.aws_eip.nat[0]
      id:                                   <computed>
      allocation_id:                        <computed>
      association_id:                       <computed>
      domain:                               <computed>
      instance:                             "${element(aws_instance.nat.*.id, count.index)}"
      network_interface:                    <computed>
      private_ip:                           <computed>
      public_ip:                            <computed>
      public_ipv4_pool:                     <computed>
      vpc:                                  "true"

  + module.vpc.aws_eip.nat[1]
      id:                                   <computed>
      allocation_id:                        <computed>
      association_id:                       <computed>
      domain:                               <computed>
      instance:                             "${element(aws_instance.nat.*.id, count.index)}"
      network_interface:                    <computed>
      private_ip:                           <computed>
      public_ip:                            <computed>
      public_ipv4_pool:                     <computed>
      vpc:                                  "true"

  + module.vpc.aws_instance.nat[0]
      id:                                   <computed>
      ami:                                  "ami-79eca201"
      arn:                                  <computed>
      associate_public_ip_address:          <computed>
      availability_zone:                    "us-west-2a"
      cpu_core_count:                       <computed>
      cpu_threads_per_core:                 <computed>
      ebs_block_device.#:                   <computed>
      ephemeral_block_device.#:             <computed>
      get_password_data:                    "false"
      host_id:                              <computed>
      instance_state:                       <computed>
      instance_type:                        "t2.micro"
      ipv6_address_count:                   <computed>
      ipv6_addresses.#:                     <computed>
      key_name:                             <computed>
      network_interface.#:                  <computed>
      network_interface_id:                 <computed>
      password_data:                        <computed>
      placement_group:                      <computed>
      primary_network_interface_id:         <computed>
      private_dns:                          <computed>
      private_ip:                           <computed>
      public_dns:                           <computed>
      public_ip:                            <computed>
      root_block_device.#:                  <computed>
      security_groups.#:                    <computed>
      source_dest_check:                    "true"
      subnet_id:                            "${element(module.vpc.public_subnets, count.index)}"
      tags.%:                               "2"
      tags.Environment:                     "dev"
      tags.Project:                         "project"
      tenancy:                              <computed>
      volume_tags.%:                        <computed>
      vpc_security_group_ids.#:             <computed>

  + module.vpc.aws_instance.nat[1]
      id:                                   <computed>
      ami:                                  "ami-79eca201"
      arn:                                  <computed>
      associate_public_ip_address:          <computed>
      availability_zone:                    "us-west-2b"
      cpu_core_count:                       <computed>
      cpu_threads_per_core:                 <computed>
      ebs_block_device.#:                   <computed>
      ephemeral_block_device.#:             <computed>
      get_password_data:                    "false"
      host_id:                              <computed>
      instance_state:                       <computed>
      instance_type:                        "t2.micro"
      ipv6_address_count:                   <computed>
      ipv6_addresses.#:                     <computed>
      key_name:                             <computed>
      network_interface.#:                  <computed>
      network_interface_id:                 <computed>
      password_data:                        <computed>
      placement_group:                      <computed>
      primary_network_interface_id:         <computed>
      private_dns:                          <computed>
      private_ip:                           <computed>
      public_dns:                           <computed>
      public_ip:                            <computed>
      root_block_device.#:                  <computed>
      security_groups.#:                    <computed>
      source_dest_check:                    "true"
      subnet_id:                            "${element(module.vpc.public_subnets, count.index)}"
      tags.%:                               "2"
      tags.Environment:                     "dev"
      tags.Project:                         "project"
      tenancy:                              <computed>
      volume_tags.%:                        <computed>
      vpc_security_group_ids.#:             <computed>

  + module.vpc.aws_route.private_nat_ec2[0]
      id:                                   <computed>
      destination_cidr_block:               "0.0.0.0/0"
      destination_prefix_list_id:           <computed>
      egress_only_gateway_id:               <computed>
      gateway_id:                           <computed>
      instance_id:                          "${element(aws_instance.nat.*.id, count.index)}"
      instance_owner_id:                    <computed>
      nat_gateway_id:                       <computed>
      network_interface_id:                 <computed>
      origin:                               <computed>
      route_table_id:                       "${element(module.vpc.private_route_table_ids, count.index)}"
      state:                                <computed>

  + module.vpc.aws_route.private_nat_ec2[1]
      id:                                   <computed>
      destination_cidr_block:               "0.0.0.0/0"
      destination_prefix_list_id:           <computed>
      egress_only_gateway_id:               <computed>
      gateway_id:                           <computed>
      instance_id:                          "${element(aws_instance.nat.*.id, count.index)}"
      instance_owner_id:                    <computed>
      nat_gateway_id:                       <computed>
      network_interface_id:                 <computed>
      origin:                               <computed>
      route_table_id:                       "${element(module.vpc.private_route_table_ids, count.index)}"
      state:                                <computed>

  + module.vpc.module.vpc.aws_internet_gateway.this
      id:                                   <computed>
      owner_id:                             <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev"
      tags.Project:                         "project"
      vpc_id:                               "${local.vpc_id}"

  + module.vpc.module.vpc.aws_route.public_internet_gateway
      id:                                   <computed>
      destination_cidr_block:               "0.0.0.0/0"
      destination_prefix_list_id:           <computed>
      egress_only_gateway_id:               <computed>
      gateway_id:                           "${aws_internet_gateway.this.id}"
      instance_id:                          <computed>
      instance_owner_id:                    <computed>
      nat_gateway_id:                       <computed>
      network_interface_id:                 <computed>
      origin:                               <computed>
      route_table_id:                       "${aws_route_table.public.id}"
      state:                                <computed>

  + module.vpc.module.vpc.aws_route_table.private[0]
      id:                                   <computed>
      owner_id:                             <computed>
      propagating_vgws.#:                   <computed>
      route.#:                              <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev-private-us-west-2a"
      tags.Project:                         "project"
      vpc_id:                               "${local.vpc_id}"

  + module.vpc.module.vpc.aws_route_table.private[1]
      id:                                   <computed>
      owner_id:                             <computed>
      propagating_vgws.#:                   <computed>
      route.#:                              <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev-private-us-west-2b"
      tags.Project:                         "project"
      vpc_id:                               "${local.vpc_id}"

  + module.vpc.module.vpc.aws_route_table.public
      id:                                   <computed>
      owner_id:                             <computed>
      propagating_vgws.#:                   <computed>
      route.#:                              <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev-public"
      tags.Project:                         "project"
      vpc_id:                               "${local.vpc_id}"

  + module.vpc.module.vpc.aws_route_table_association.private[0]
      id:                                   <computed>
      route_table_id:                       "${element(aws_route_table.private.*.id, (var.single_nat_gateway ? 0 : count.index))}"
      subnet_id:                            "${element(aws_subnet.private.*.id, count.index)}"

  + module.vpc.module.vpc.aws_route_table_association.private[1]
      id:                                   <computed>
      route_table_id:                       "${element(aws_route_table.private.*.id, (var.single_nat_gateway ? 0 : count.index))}"
      subnet_id:                            "${element(aws_subnet.private.*.id, count.index)}"

  + module.vpc.module.vpc.aws_route_table_association.public[0]
      id:                                   <computed>
      route_table_id:                       "${aws_route_table.public.id}"
      subnet_id:                            "${element(aws_subnet.public.*.id, count.index)}"

  + module.vpc.module.vpc.aws_route_table_association.public[1]
      id:                                   <computed>
      route_table_id:                       "${aws_route_table.public.id}"
      subnet_id:                            "${element(aws_subnet.public.*.id, count.index)}"

  + module.vpc.module.vpc.aws_subnet.private[0]
      id:                                   <computed>
      arn:                                  <computed>
      assign_ipv6_address_on_creation:      "false"
      availability_zone:                    "us-west-2a"
      availability_zone_id:                 <computed>
      cidr_block:                           "10.10.1.0/24"
      ipv6_cidr_block:                      <computed>
      ipv6_cidr_block_association_id:       <computed>
      map_public_ip_on_launch:              "false"
      owner_id:                             <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev-private-us-west-2a"
      tags.Project:                         "project"
      vpc_id:                               "${local.vpc_id}"

  + module.vpc.module.vpc.aws_subnet.private[1]
      id:                                   <computed>
      arn:                                  <computed>
      assign_ipv6_address_on_creation:      "false"
      availability_zone:                    "us-west-2b"
      availability_zone_id:                 <computed>
      cidr_block:                           "10.10.2.0/24"
      ipv6_cidr_block:                      <computed>
      ipv6_cidr_block_association_id:       <computed>
      map_public_ip_on_launch:              "false"
      owner_id:                             <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev-private-us-west-2b"
      tags.Project:                         "project"
      vpc_id:                               "${local.vpc_id}"

  + module.vpc.module.vpc.aws_subnet.public[0]
      id:                                   <computed>
      arn:                                  <computed>
      assign_ipv6_address_on_creation:      "false"
      availability_zone:                    "us-west-2a"
      availability_zone_id:                 <computed>
      cidr_block:                           "10.10.11.0/24"
      ipv6_cidr_block:                      <computed>
      ipv6_cidr_block_association_id:       <computed>
      map_public_ip_on_launch:              "true"
      owner_id:                             <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev-public-us-west-2a"
      tags.Project:                         "project"
      vpc_id:                               "${local.vpc_id}"

  + module.vpc.module.vpc.aws_subnet.public[1]
      id:                                   <computed>
      arn:                                  <computed>
      assign_ipv6_address_on_creation:      "false"
      availability_zone:                    "us-west-2b"
      availability_zone_id:                 <computed>
      cidr_block:                           "10.10.12.0/24"
      ipv6_cidr_block:                      <computed>
      ipv6_cidr_block_association_id:       <computed>
      map_public_ip_on_launch:              "true"
      owner_id:                             <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev-public-us-west-2b"
      tags.Project:                         "project"
      vpc_id:                               "${local.vpc_id}"

  + module.vpc.module.vpc.aws_vpc.this
      id:                                   <computed>
      arn:                                  <computed>
      assign_generated_ipv6_cidr_block:     "false"
      cidr_block:                           "10.10.0.0/16"
      default_network_acl_id:               <computed>
      default_route_table_id:               <computed>
      default_security_group_id:            <computed>
      dhcp_options_id:                      <computed>
      enable_classiclink:                   <computed>
      enable_classiclink_dns_support:       <computed>
      enable_dns_hostnames:                 "false"
      enable_dns_support:                   "true"
      instance_tenancy:                     "default"
      ipv6_association_id:                  <computed>
      ipv6_cidr_block:                      <computed>
      main_route_table_id:                  <computed>
      owner_id:                             <computed>
      tags.%:                               "3"
      tags.Environment:                     "dev"
      tags.Name:                            "project-dev"
      tags.Project:                         "project"


Plan: 21 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
```
apply logs:
```
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.vpc.module.vpc.aws_vpc.this: Creating...
  arn:                              "" => "<computed>"
  assign_generated_ipv6_cidr_block: "" => "false"
  cidr_block:                       "" => "10.10.0.0/16"
  default_network_acl_id:           "" => "<computed>"
  default_route_table_id:           "" => "<computed>"
  default_security_group_id:        "" => "<computed>"
  dhcp_options_id:                  "" => "<computed>"
  enable_classiclink:               "" => "<computed>"
  enable_classiclink_dns_support:   "" => "<computed>"
  enable_dns_hostnames:             "" => "false"
  enable_dns_support:               "" => "true"
  instance_tenancy:                 "" => "default"
  ipv6_association_id:              "" => "<computed>"
  ipv6_cidr_block:                  "" => "<computed>"
  main_route_table_id:              "" => "<computed>"
  owner_id:                         "" => "<computed>"
  tags.%:                           "" => "3"
  tags.Environment:                 "" => "dev"
  tags.Name:                        "" => "project-dev"
  tags.Project:                     "" => "project"
module.vpc.vpc.aws_vpc.this: Still creating... (10s elapsed)
module.vpc.module.vpc.aws_vpc.this: Creation complete after 15s (ID: vpc-072436456b5e62c52)
module.vpc.module.vpc.aws_subnet.public[0]: Creating...
  arn:                             "" => "<computed>"
  assign_ipv6_address_on_creation: "" => "false"
  availability_zone:               "" => "us-west-2a"
  availability_zone_id:            "" => "<computed>"
  cidr_block:                      "" => "10.10.11.0/24"
  ipv6_cidr_block:                 "" => "<computed>"
  ipv6_cidr_block_association_id:  "" => "<computed>"
  map_public_ip_on_launch:         "" => "true"
  owner_id:                        "" => "<computed>"
  tags.%:                          "" => "3"
  tags.Environment:                "" => "dev"
  tags.Name:                       "" => "project-dev-public-us-west-2a"
  tags.Project:                    "" => "project"
  vpc_id:                          "" => "vpc-072436456b5e62c52"
module.vpc.module.vpc.aws_subnet.public[1]: Creating...
  arn:                             "" => "<computed>"
  assign_ipv6_address_on_creation: "" => "false"
  availability_zone:               "" => "us-west-2b"
  availability_zone_id:            "" => "<computed>"
  cidr_block:                      "" => "10.10.12.0/24"
  ipv6_cidr_block:                 "" => "<computed>"
  ipv6_cidr_block_association_id:  "" => "<computed>"
  map_public_ip_on_launch:         "" => "true"
  owner_id:                        "" => "<computed>"
  tags.%:                          "" => "3"
  tags.Environment:                "" => "dev"
  tags.Name:                       "" => "project-dev-public-us-west-2b"
  tags.Project:                    "" => "project"
  vpc_id:                          "" => "vpc-072436456b5e62c52"
module.vpc.module.vpc.aws_internet_gateway.this: Creating...
  owner_id:         "" => "<computed>"
  tags.%:           "0" => "3"
  tags.Environment: "" => "dev"
  tags.Name:        "" => "project-dev"
  tags.Project:     "" => "project"
  vpc_id:           "" => "vpc-072436456b5e62c52"
module.vpc.module.vpc.aws_route_table.public: Creating...
  owner_id:           "" => "<computed>"
  propagating_vgws.#: "" => "<computed>"
  route.#:            "" => "<computed>"
  tags.%:             "" => "3"
  tags.Environment:   "" => "dev"
  tags.Name:          "" => "project-dev-public"
  tags.Project:       "" => "project"
  vpc_id:             "" => "vpc-072436456b5e62c52"
module.vpc.module.vpc.aws_subnet.private[0]: Creating...
  arn:                             "" => "<computed>"
  assign_ipv6_address_on_creation: "" => "false"
  availability_zone:               "" => "us-west-2a"
  availability_zone_id:            "" => "<computed>"
  cidr_block:                      "" => "10.10.1.0/24"
  ipv6_cidr_block:                 "" => "<computed>"
  ipv6_cidr_block_association_id:  "" => "<computed>"
  map_public_ip_on_launch:         "" => "false"
  owner_id:                        "" => "<computed>"
  tags.%:                          "" => "3"
  tags.Environment:                "" => "dev"
  tags.Name:                       "" => "project-dev-private-us-west-2a"
  tags.Project:                    "" => "project"
  vpc_id:                          "" => "vpc-072436456b5e62c52"
module.vpc.module.vpc.aws_subnet.private[1]: Creating...
  arn:                             "" => "<computed>"
  assign_ipv6_address_on_creation: "" => "false"
  availability_zone:               "" => "us-west-2b"
  availability_zone_id:            "" => "<computed>"
  cidr_block:                      "" => "10.10.2.0/24"
  ipv6_cidr_block:                 "" => "<computed>"
  ipv6_cidr_block_association_id:  "" => "<computed>"
  map_public_ip_on_launch:         "" => "false"
  owner_id:                        "" => "<computed>"
  tags.%:                          "" => "3"
  tags.Environment:                "" => "dev"
  tags.Name:                       "" => "project-dev-private-us-west-2b"
  tags.Project:                    "" => "project"
  vpc_id:                          "" => "vpc-072436456b5e62c52"
module.vpc.aws_default_security_group.assign-name: Creating...
  arn:                                  "" => "<computed>"
  egress.#:                             "" => "1"
  egress.482069346.cidr_blocks.#:       "" => "1"
  egress.482069346.cidr_blocks.0:       "" => "0.0.0.0/0"
  egress.482069346.description:         "" => ""
  egress.482069346.from_port:           "" => "0"
  egress.482069346.ipv6_cidr_blocks.#:  "" => "0"
  egress.482069346.prefix_list_ids.#:   "" => "0"
  egress.482069346.protocol:            "" => "-1"
  egress.482069346.security_groups.#:   "" => "0"
  egress.482069346.self:                "" => "false"
  egress.482069346.to_port:             "" => "0"
  ingress.#:                            "" => "1"
  ingress.753360330.cidr_blocks.#:      "" => "0"
  ingress.753360330.description:        "" => ""
  ingress.753360330.from_port:          "" => "0"
  ingress.753360330.ipv6_cidr_blocks.#: "" => "0"
  ingress.753360330.prefix_list_ids.#:  "" => "0"
  ingress.753360330.protocol:           "" => "-1"
  ingress.753360330.security_groups.#:  "" => "0"
  ingress.753360330.self:               "" => "true"
  ingress.753360330.to_port:            "" => "0"
  name:                                 "" => "<computed>"
  owner_id:                             "" => "<computed>"
  revoke_rules_on_delete:               "" => "false"
  tags.%:                               "" => "3"
  tags.Environment:                     "" => "dev"
  tags.Name:                            "" => "project-dev"
  tags.Project:                         "" => "project"
  vpc_id:                               "" => "vpc-072436456b5e62c52"
module.vpc.module.vpc.aws_route_table.private[0]: Creating...
  owner_id:           "" => "<computed>"
  propagating_vgws.#: "" => "<computed>"
  route.#:            "" => "<computed>"
  tags.%:             "" => "3"
  tags.Environment:   "" => "dev"
  tags.Name:          "" => "project-dev-private-us-west-2a"
  tags.Project:       "" => "project"
  vpc_id:             "" => "vpc-072436456b5e62c52"
module.vpc.module.vpc.aws_route_table.private[1]: Creating...
  owner_id:           "" => "<computed>"
  propagating_vgws.#: "" => "<computed>"
  route.#:            "" => "<computed>"
  tags.%:             "" => "3"
  tags.Environment:   "" => "dev"
  tags.Name:          "" => "project-dev-private-us-west-2b"
  tags.Project:       "" => "project"
  vpc_id:             "" => "vpc-072436456b5e62c52"
module.vpc.module.vpc.aws_route_table.public: Creation complete after 4s (ID: rtb-09e7905cc89b29c74)
module.vpc.module.vpc.aws_route_table.private[0]: Creation complete after 5s (ID: rtb-0ea52fc52a1f9425c)
module.vpc.module.vpc.aws_route_table.private[1]: Creation complete after 5s (ID: rtb-08bec7dc8683fd395)
module.vpc.module.vpc.aws_subnet.private[1]: Creation complete after 5s (ID: subnet-0133a1436890f999e)
module.vpc.module.vpc.aws_subnet.private[0]: Creation complete after 5s (ID: subnet-05b7ea9d96c7d13d6)
module.vpc.module.vpc.aws_route_table_association.private[1]: Creating...
  route_table_id: "" => "rtb-08bec7dc8683fd395"
  subnet_id:      "" => "subnet-0133a1436890f999e"
module.vpc.module.vpc.aws_route_table_association.private[0]: Creating...
  route_table_id: "" => "rtb-0ea52fc52a1f9425c"
  subnet_id:      "" => "subnet-05b7ea9d96c7d13d6"
module.vpc.module.vpc.aws_subnet.public[1]: Creation complete after 6s (ID: subnet-09d463b1cd1aaa57d)
module.vpc.module.vpc.aws_subnet.public[0]: Creation complete after 6s (ID: subnet-029e7db1ab668d65c)
module.vpc.module.vpc.aws_route_table_association.public[1]: Creating...
  route_table_id: "" => "rtb-09e7905cc89b29c74"
  subnet_id:      "" => "subnet-09d463b1cd1aaa57d"
module.vpc.module.vpc.aws_route_table_association.public[0]: Creating...
  route_table_id: "" => "rtb-09e7905cc89b29c74"
  subnet_id:      "" => "subnet-029e7db1ab668d65c"
module.vpc.module.vpc.aws_route_table_association.private[0]: Creation complete after 1s (ID: rtbassoc-0a186b050c6040e9d)
module.vpc.module.vpc.aws_route_table_association.private[1]: Creation complete after 1s (ID: rtbassoc-0287855341dec5e56)
module.vpc.module.vpc.aws_internet_gateway.this: Creation complete after 7s (ID: igw-0ced41a3362e44ce3)
module.vpc.module.vpc.aws_route.public_internet_gateway: Creating...
  destination_cidr_block:     "" => "0.0.0.0/0"
  destination_prefix_list_id: "" => "<computed>"
  egress_only_gateway_id:     "" => "<computed>"
  gateway_id:                 "" => "igw-0ced41a3362e44ce3"
  instance_id:                "" => "<computed>"
  instance_owner_id:          "" => "<computed>"
  nat_gateway_id:             "" => "<computed>"
  network_interface_id:       "" => "<computed>"
  origin:                     "" => "<computed>"
  route_table_id:             "" => "rtb-09e7905cc89b29c74"
  state:                      "" => "<computed>"
module.vpc.module.vpc.aws_route_table_association.public[1]: Creation complete after 1s (ID: rtbassoc-0fd6e8074ad851705)
module.vpc.module.vpc.aws_route_table_association.public[0]: Creation complete after 1s (ID: rtbassoc-00114155357d8a344)
module.vpc.module.vpc.aws_route.public_internet_gateway: Creation complete after 2s (ID: r-rtb-09e7905cc89b29c741080289494)
module.vpc.aws_default_security_group.assign-name: Still creating... (10s elapsed)
module.vpc.aws_default_security_group.assign-name: Creation complete after 18s (ID: sg-05ad340907f5ca6c3)
module.vpc.aws_instance.nat[0]: Creating...
  ami:                               "" => "ami-79eca201"
  arn:                               "" => "<computed>"
  associate_public_ip_address:       "" => "<computed>"
  availability_zone:                 "" => "us-west-2a"
  cpu_core_count:                    "" => "<computed>"
  cpu_threads_per_core:              "" => "<computed>"
  ebs_block_device.#:                "" => "<computed>"
  ephemeral_block_device.#:          "" => "<computed>"
  get_password_data:                 "" => "false"
  host_id:                           "" => "<computed>"
  instance_state:                    "" => "<computed>"
  instance_type:                     "" => "t2.micro"
  ipv6_address_count:                "" => "<computed>"
  ipv6_addresses.#:                  "" => "<computed>"
  key_name:                          "" => "<computed>"
  network_interface.#:               "" => "<computed>"
  network_interface_id:              "" => "<computed>"
  password_data:                     "" => "<computed>"
  placement_group:                   "" => "<computed>"
  primary_network_interface_id:      "" => "<computed>"
  private_dns:                       "" => "<computed>"
  private_ip:                        "" => "<computed>"
  public_dns:                        "" => "<computed>"
  public_ip:                         "" => "<computed>"
  root_block_device.#:               "" => "<computed>"
  security_groups.#:                 "" => "<computed>"
  source_dest_check:                 "" => "true"
  subnet_id:                         "" => "subnet-029e7db1ab668d65c"
  tags.%:                            "" => "2"
  tags.Environment:                  "" => "dev"
  tags.Project:                      "" => "project"
  tenancy:                           "" => "<computed>"
  volume_tags.%:                     "" => "<computed>"
  vpc_security_group_ids.#:          "" => "1"
  vpc_security_group_ids.3169845296: "" => "sg-05ad340907f5ca6c3"
module.vpc.aws_instance.nat[1]: Creating...
  ami:                               "" => "ami-79eca201"
  arn:                               "" => "<computed>"
  associate_public_ip_address:       "" => "<computed>"
  availability_zone:                 "" => "us-west-2b"
  cpu_core_count:                    "" => "<computed>"
  cpu_threads_per_core:              "" => "<computed>"
  ebs_block_device.#:                "" => "<computed>"
  ephemeral_block_device.#:          "" => "<computed>"
  get_password_data:                 "" => "false"
  host_id:                           "" => "<computed>"
  instance_state:                    "" => "<computed>"
  instance_type:                     "" => "t2.micro"
  ipv6_address_count:                "" => "<computed>"
  ipv6_addresses.#:                  "" => "<computed>"
  key_name:                          "" => "<computed>"
  network_interface.#:               "" => "<computed>"
  network_interface_id:              "" => "<computed>"
  password_data:                     "" => "<computed>"
  placement_group:                   "" => "<computed>"
  primary_network_interface_id:      "" => "<computed>"
  private_dns:                       "" => "<computed>"
  private_ip:                        "" => "<computed>"
  public_dns:                        "" => "<computed>"
  public_ip:                         "" => "<computed>"
  root_block_device.#:               "" => "<computed>"
  security_groups.#:                 "" => "<computed>"
  source_dest_check:                 "" => "true"
  subnet_id:                         "" => "subnet-09d463b1cd1aaa57d"
  tags.%:                            "" => "2"
  tags.Environment:                  "" => "dev"
  tags.Project:                      "" => "project"
  tenancy:                           "" => "<computed>"
  volume_tags.%:                     "" => "<computed>"
  vpc_security_group_ids.#:          "" => "1"
  vpc_security_group_ids.3169845296: "" => "sg-05ad340907f5ca6c3"
module.vpc.aws_instance.nat.1: Still creating... (10s elapsed)
module.vpc.aws_instance.nat.0: Still creating... (10s elapsed)
module.vpc.aws_instance.nat.0: Still creating... (20s elapsed)
module.vpc.aws_instance.nat.1: Still creating... (20s elapsed)
module.vpc.aws_instance.nat.0: Still creating... (30s elapsed)
module.vpc.aws_instance.nat.1: Still creating... (30s elapsed)
module.vpc.aws_instance.nat[1]: Creation complete after 34s (ID: i-01827fac7dec2143c)
module.vpc.aws_instance.nat[0]: Creation complete after 34s (ID: i-0772dbac9a393432d)
module.vpc.aws_eip.nat[0]: Creating...
  allocation_id:     "" => "<computed>"
  association_id:    "" => "<computed>"
  domain:            "" => "<computed>"
  instance:          "" => "i-0772dbac9a393432d"
  network_interface: "" => "<computed>"
  private_ip:        "" => "<computed>"
  public_ip:         "" => "<computed>"
  public_ipv4_pool:  "" => "<computed>"
  vpc:               "" => "true"
module.vpc.aws_eip.nat[1]: Creating...
  allocation_id:     "" => "<computed>"
  association_id:    "" => "<computed>"
  domain:            "" => "<computed>"
  instance:          "" => "i-01827fac7dec2143c"
  network_interface: "" => "<computed>"
  private_ip:        "" => "<computed>"
  public_ip:         "" => "<computed>"
  public_ipv4_pool:  "" => "<computed>"
  vpc:               "" => "true"
module.vpc.aws_route.private_nat_ec2[1]: Creating...
  destination_cidr_block:     "" => "0.0.0.0/0"
  destination_prefix_list_id: "" => "<computed>"
  egress_only_gateway_id:     "" => "<computed>"
  gateway_id:                 "" => "<computed>"
  instance_id:                "" => "i-01827fac7dec2143c"
  instance_owner_id:          "" => "<computed>"
  nat_gateway_id:             "" => "<computed>"
  network_interface_id:       "" => "<computed>"
  origin:                     "" => "<computed>"
  route_table_id:             "" => "rtb-08bec7dc8683fd395"
  state:                      "" => "<computed>"
module.vpc.aws_route.private_nat_ec2[0]: Creating...
  destination_cidr_block:     "" => "0.0.0.0/0"
  destination_prefix_list_id: "" => "<computed>"
  egress_only_gateway_id:     "" => "<computed>"
  gateway_id:                 "" => "<computed>"
  instance_id:                "" => "i-0772dbac9a393432d"
  instance_owner_id:          "" => "<computed>"
  nat_gateway_id:             "" => "<computed>"
  network_interface_id:       "" => "<computed>"
  origin:                     "" => "<computed>"
  route_table_id:             "" => "rtb-0ea52fc52a1f9425c"
  state:                      "" => "<computed>"
module.vpc.aws_route.private_nat_ec2[0]: Creation complete after 2s (ID: r-rtb-0ea52fc52a1f9425c1080289494)
module.vpc.aws_route.private_nat_ec2[1]: Creation complete after 2s (ID: r-rtb-08bec7dc8683fd3951080289494)
module.vpc.aws_eip.nat[1]: Creation complete after 4s (ID: eipalloc-0c9bc5b43c72ca189)
module.vpc.aws_eip.nat[0]: Creation complete after 4s (ID: eipalloc-0fdcf5e6c8ff8e2dd)

Apply complete! Resources: 21 added, 0 changed, 0 destroyed.
```

**Pull request merged**